test/logflare/sql/dialect_transformer/clickhouse_test.exs

```diff
@@ -0,0 +1,66 @@
+defmodule Logflare.Sql.DialectTransformer.ClickhouseTest do
+  use Logflare.DataCase
+
+  alias Logflare.Sql.DialectTransformer.Clickhouse
+  alias Logflare.Backends.Adaptor.ClickhouseAdaptor
+
+  describe "quote_style/0" do
+    test "returns nil for ClickHouse" do
+      assert Clickhouse.quote_style() == nil
+    end
+  end
+
+  describe "dialect/0" do
+    test "returns clickhouse string" do
+      assert Clickhouse.dialect() == "clickhouse"
+    end
+  end
+
+  describe "transform_source_name/2" do
+    test "delegates to ClickhouseAdaptor.clickhouse_ingest_table_name/1" do
+      user = insert(:user)
+      source = insert(:source, name: "test_source", user: user)
+
+      data = %{
+        sources: [source],
+        dialect: "clickhouse"
+      }
+
+      result = Clickhouse.transform_source_name("test_source", data)
+      expected = ClickhouseAdaptor.clickhouse_ingest_table_name(source)
+
+      assert result == expected
+    end
+
+    test "finds correct source by name from multiple sources" do
+      user = insert(:user)
+      source1 = insert(:source, name: "source_one", user: user)
+      source2 = insert(:source, name: "source_two", user: user)
+
+      data = %{
+        sources: [source1, source2],
+        dialect: "clickhouse"
+      }
+
+      result = Clickhouse.transform_source_name("source_two", data)
+      expected = ClickhouseAdaptor.clickhouse_ingest_table_name(source2)
+
+      assert result == expected
+    end
+
+    test "handles source with special characters in token" do
+      user = insert(:user)
+      source = insert(:source, name: "special_source", user: user)
```

Comment:
we can consider building these structs instead of inserting so that we can make the tests async (since there is nothing repo related for the sql stuff)
low priority